data_structures/rtree.cpp

```diff
@@ -0,0 +1,168 @@
+// rtree.cpp
+
+#include <iostream>
+#include <memory>
+
+#include "rtree.hpp"
+
+using coord_t = spatial::coord_t;
```

Comment:
```suggestion
using coord_t = spatial::coord_t;
using area_t = coord_t;
```
In `choose_branch()`, it might make for more semantic typing to introduce a new alias here.